gui/kitchen-info.lua

```diff
@@ -0,0 +1,404 @@
+        hidden = true  -- This column is hidden because the only plant with BAG_ITEM reaction is Quarry Bush, which is neither cookable nor brewable so it never appears in the Kitchen screen
+    },
+    {
+        label = "Thread",
+        maxValueLength = 6,
+        minValueLeftPadding = 0,
+        emptyValueLeftPadding = 1,
+        activeOnPages = { 'VegetablesFruitLeaves' }
+    },
+    {
+        label = "Mill",
+        maxValueLength = 5,
+        minValueLeftPadding = 0,
+        activeOnPages = { 'VegetablesFruitLeaves', 'Seeds' }
+    },
+    --[[
+    -- Unimplemented
+    {
+        label = "Press",
+        maxValueLength = 5,
+        minValueLeftPadding = 0,
+        activeOnPages = { 'VegetablesFruitLeaves', 'Seeds' }
+    },
+    --]]
+    {
+        label = "EatRaw",
+        maxValueLength = 4,
+        minValueLeftPadding = 1,
+        emptyValueLeftPadding = 1,
+        activeOnPages = { 'VegetablesFruitLeaves', 'Seeds' }
+    }
+}
+
+local COLUMNS_BY_LABEL = {}
+
+for _, column in ipairs(COLUMNS) do
+    column.activeOnPages = utils.invert(column.activeOnPages)
+    column.minLength = math.max(#column.label, column.minValueLeftPadding + column.maxValueLength)
+
+    COLUMNS_BY_LABEL[column.label] = column
+end
+
+function COLUMNS_BY_LABEL.Extract.getColumnValue(item_type, material, plant_raw)
+    if (item_type == df.item_type.PLANT) then
+        if (plant_raw.flags.EXTRACT_BARREL) then
+            return "Barrel"
+        elseif (plant_raw.flags.EXTRACT_VIAL) then
+            return " Vial"
+        elseif (plant_raw.flags.EXTRACT_STILL_VIAL) then
+            return " Still"
+        end
+    end
+    return false
+end
+
+-- This column is hidden because the only plant with BAG_ITEM reaction is Quarry Bush, which is neither cookable nor brewable so it never appears in the Kitchen screen
+function COLUMNS_BY_LABEL.ToBag.getColumnValue(item_type, material, plant_raw)
+    if (item_type == df.item_type.PLANT) then
+        local reaction_product = material.reaction_product
+        for i, reaction_product_id in ipairs(reaction_product.id) do
+            if (reaction_product_id.value == "BAG_ITEM") then
+                local bag_item_type = reaction_product.item_type[i]
+
+                if (bag_item_type == df.item_type.PLANT_GROWTH) then
+                    local bag_item_subtype = reaction_product.item_subtype[i]
+                    local bag_mat_type  = reaction_product.material.mat_type[i]
+                    local bag_mat_index = reaction_product.material.mat_index[i]
+                    local bag_matinfo = dfhack.matinfo.decode(bag_mat_type, bag_mat_index)
+                    local bag_mat_plant_raw = bag_matinfo.plant
+                    local bag_mat_plant_growth = bag_mat_plant_raw.growths[bag_item_subtype]
+
+                    if (bag_mat_plant_growth.name_plural:sub(-6) == "leaves") then
+                        return "Leaves"
+                    end
```

Comment:
Is this really the best way to implement this test?

Reply:
Again, user experience.
"Leaves" in the name is what the user sees.
LEAVES in growth type, or LEAF material based on LEAF_TEMPLATE material template are all internal tags that can be named whatever and work the same.

(And tbh it is mostly to support vanilla Quarry Bush, which is the only vanilla plant with this reaction. Modded items will just get "Other" so user have to look it up)